dot/network/discovery.go

```diff
@@ -92,6 +92,9 @@ func (d *discovery) waitForPeers() (peers []peer.AddrInfo, err error) {
 
 // start creates the DHT.
 func (d *discovery) start() error {
+	// this basically only works with enabled mDNS which is used only for local test setups. Without bootnodes kademilia
+	// would not bee able to connect to any peers and mDNS is used to find peers in local network.
+	// TODO: should be refactored because this if is basically used for local integration test purpose
```

Comment:
Instead of waiting for peers to connect to start kad we can upgrade the kad routing table on every connection, I think that using `d.dht.{LAN/WAN}.RoutingTable().UsefulNewPeer(peerID)` should be a good option